pkg/apiproxy/ytsaurus.go

```diff
@@ -115,13 +115,12 @@ func (c *Ytsaurus) SaveClusterState(ctx context.Context, clusterState ytv1.Clust
 	return nil
 }
 
-func (c *Ytsaurus) SaveObserverGeneration(ctx context.Context, generation int64) error {
-	c.ytsaurus.Status.ObservedGeneration = generation
-	if err := c.apiProxy.UpdateStatus(ctx); err != nil {
-		return fmt.Errorf("failed to save observed generation in status: %w", err)
+func (c *Ytsaurus) SyncObservedGeneration() bool {
```

Comment:
Please add doc line to determine whenever we return true or false.

Reply:
Good point